### DISCUSSION TITLE
planar illumination source

### DISCUSSION THREAD

**Discussion by leoyala** [fangq/mcx#220] [Help]
Hello, 

I am trying to use a planar illumination source, but got confused on how to input the extent of the plane. From the implementation described here:  https://github.com/fangq/mcx/blob/master/src/mcx_const.h#L88 I see that it is defined in srcparam1, but I am not sure how to define both the position and extent of the plane with only 3 values. 

Kind regards.

**Comment by fangq** [comment]
planar source has a quadrilateral-shaped launch aperture, with the 4 corners defined by

`[srcpos, srcpos+srcparam1(1:3), srcpos+srcparam1(1:3)+srcparam2(1:3), srcpos+srcparam2(1:3)]`

https://github.com/fangq/mcx/blob/v2024.2/mcxlab/mcxlab.m#L196-L197

so, if you specify srcpos=[10,12,0], srcparam1=[10,0,0] and srcparam2=[0,20,0], then the src area covers a square between x=[10,20] and y=[12,32].
